seed/static/seed/js/controllers/sample_data_modal_controller.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * :copyright (c) 2014 - 2021, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.
+ * :author
+ */
+angular.module('BE.seed.controller.sample_data_modal', [])
+  .controller('sample_data_modal_controller', [
+    '$scope',
+    '$state',
+    '$uibModalInstance',
+    'Notification',
+    'inventory_service',
+    'organization_service',
+    'organization',
+    'cycle',
+    'profiles',
+    function (
+      $scope,
+      $state,
+      $uibModalInstance,
+      Notification,
+      inventory_service,
+      organization_service,
+      organization,
+      cycle,
+      profiles
+    ) {
+      $scope.inProgress = false;
+      $scope.hasData = cycle.num_properties > 0 || cycle.num_taxlots > 0;
+
+      $scope.continue = function () {
+        $scope.inProgress = true;
+
+        // Create column list profile if it doesn't exist
+        let foundProfile = profiles.find(({name, profile_location}) => name === 'Auto-Populate' && profile_location === 'List View Profile');
+        let profilePromise;
+        if (foundProfile) {
+          profilePromise = new Promise(resolve => resolve(foundProfile));
+        } else {
+          profilePromise = inventory_service.new_column_list_profile({
+            name: 'Auto-Populate',
+            profile_location: 'List View Profile',
+            inventory_type: 'Property',
+            columns: []
+          });
+        }
+
+        profilePromise.then(profile => {
+          inventory_service.save_last_profile(profile.id, 'properties');
+          inventory_service.save_last_cycle(cycle.id)
+
+          return inventory_service.get_property_columns().then(columns => {
+            return organization_service.insert_sample_data(organization.org_id).then(() => {
+              return inventory_service.get_properties(1, undefined, cycle, -1).then(inventory => {
+                const visibleColumns = findPopulatedColumns(columns, inventory.results);
+                const profileId = profile.id;
+                profile = _.omit(profile, 'id');
+                profile.columns = visibleColumns;
+                return inventory_service.update_column_list_profile(profileId, profile);
+              });
```

Comment:
A high level comment describing the overview of this process might be helpful.

Also, seems like we're doing a lot of work on the frontend here which I would assume we could do in the endpoint?